CRM/Event/Form/Participant.php

```diff
@@ -927,7 +928,6 @@ public function submit($params) {
       $this->_params = $this->prepareParamsForPaymentProcessor($this->_params);
       $this->_params['amount'] = $params['fee_amount'];
       $this->_params['amount_level'] = $params['amount_level'];
-      $this->_params['currencyID'] = $config->defaultCurrency;
```

Comment:
@eileenmcnaughton How is this working with multiple currencies? Shouldn't this actually be passing the actual currency (of the event page) through and not the system default?

Reply:
@mlutfy yeah - not too sure - it is unchanged in this PR other than switching away from using the override class. But, it doesn't seem to be handled well & I'm not sure all the stuff that would need to be changed to fix it (given it could be edit mode)

What this PR does do is make it clear that it is ALWAYS based on the setting in this form - rather than giving the illustion that the value coming out of `$params` actually has some validity.

Perhaps a follow up would be to extract to `getCurrency()` which would at least be a place to comment on the short-coming?

Reply:
ok, I'll merge this then.

Reply:
@mattwire I created a follow up https://github.com/civicrm/civicrm-core/pull/28289